addons/account/models/account_payment.py

```diff
@@ -489,187 +674,35 @@ def post(self):
                             sequence_code = 'account.payment.supplier.refund'
                         if rec.payment_type == 'outbound':
                             sequence_code = 'account.payment.supplier.invoice'
-                rec.name = self.env['ir.sequence'].with_context(ir_sequence_date=rec.payment_date).next_by_code(sequence_code)
+                rec.name = self.env['ir.sequence'].next_by_code(sequence_code, sequence_date=rec.payment_date)
                 if not rec.name and rec.payment_type != 'transfer':
                     raise UserError(_("You have to define a sequence for %s in your company.") % (sequence_code,))
 
-            # Create the journal entry
-            amount = rec.amount * (rec.payment_type in ('outbound', 'transfer') and 1 or -1)
-            move = rec._create_payment_entry(amount)
+            moves = AccountMove.create(rec._prepare_payment_moves())
```

Comment:
It would be better if you put the create out of the for loop. Your method `_prepare_payment_moves` is already well suited for `api.multi`
You can then still change the records with
```python
for move, payment in zip(moves, self):
    # write values on rec
    # do the .reconcile() things
```

Reply:
I didn't make it because:
-  You are making the assumption the result of create is always sorted on the same order as its parameter.
- A payment could generate two moves in case of an internal transfer making your zip useless.
- I refactored the whole entries generation that is quite enough for this step I think.